packages/teams-js/src/public/media.ts

```diff
@@ -434,27 +434,33 @@ export namespace media {
     protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
 
-      return Promise.resolve()
-        .then(() => {
-          throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion);
-        })
-        .then(() => {
-          const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
-
-          sendMessageToParent('media.controller', [params], (err?: SdkError) => {
-            if (callback) {
-              callback(err);
-            } else if (err) {
-              throw err;
-            }
-          });
-        })
-        .catch(err => {
-          if (callback) {
-            callback(err);
-          }
-          throw err;
-        });
+      try {
+        throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion);
+      } catch (err) {
+        const wrappedRejectedErrorFn: InputFunction<void> = () => Promise.reject(err);
+
+        return callCallbackWithSdkErrorFromPromiseAndReturnPromise(wrappedRejectedErrorFn, callback);
+      }
+
+      const params: MediaControllerParam = {
+        mediaType: this.getMediaType(),
+        mediaControllerEvent: mediaEvent,
+      };
+
+      const wrappedFunction = (): Promise<void> =>
+        new Promise(resolve =>
+          resolve(
+            sendMessageToParent('media.controller', [params], (err?: SdkError) => {
```

Comment:
I think we want to use `sendAndHandleSdkError` for this code which calls `sendMessageToParentAsync`, rather than `sendMessageToParent`